### PR TITLE
Enable/disable log instrumentations

### LINKED ISSUE #1310: Add the possiiblity to disable Microsoft.Extensions.Logging instrumentation when CLR Profiler is enabled
I think we should add support for `OTEL_DOTNET_AUTO_LOGS_DISABLED_INSTRUMENTATIONS` and  `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS`. It would require changes in native code and `integrations.json` to define the "type" of bytecode instrumentation.

The same could be done to enable metrics bytecode instrumentation.

Side note: Is it possible to have two bytecode instrumentations for the same API? So that we can have e.g. separate bytecode GraphQL instrumentation for metrics and spans.

Moreover, I am not sure if `LoggingBuilder` is a good/"unique" name/id. Maybe `MicrosoftExtensionsLogging` or `ILogger`. The second one is not so unique but I think more obvious than `LoggingBuilder`.

Related changes: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1324

**EDITED** (kudos for @RassK)

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #1310

## What

Possibility to enable/disable log instrumentations by `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS` and `OTEL_DOTNET_AUTO_LOGS_DISABLED_INSTRUMENTATIONS`.
Introduce type for bytecode instrumentation. For now, it supports `Trace` and `Log`.

Change `LoggerBuilder` instrumentation name to `ILogger` per https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1310#issuecomment-1262574246

## Tests

CI + added new tests.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ for more details.
...
 | `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS`  | Comma-separated list of metrics source instrumentations you want to enable. Set `none` to disable all metric instrumentations.                                                                                                                                  | all available instrumentations |
 | `OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS` | Comma-separated list of metrics source instrumentations you want to disable.                                                                                                                                                                                    |                                |
+| `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS`     | Comma-separated list of logs source instrumentations you want to enable. Set `none` to disable all metric instrumentations.                                                                                                                                     | all available instrumentations |
```
Comment (theletterf) [inline docs/config.md:32]:
```suggestion
| `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS`     | Comma-separated list of logs source instrumentations you want to enable. Set to `none` to disable all metric instrumentations.                                                                                                                                     | all available instrumentations |
```

**Review by theletterf — APPROVED** [review]

File: docs/config.md (hunk 1 above)
Comment (Kielek) [reply docs/config.md]: Fixed in c0871bfa4e1056981df4aa9f13feb2f48cde03a9 together with other places.

File: src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.h [hunk 2]
```diff
@@ -19,29 +19,37 @@ using json = nlohmann::json;
...
 namespace
 {
     void IntegrationFromJson(const json::value_type& src,
                          std::vector<IntegrationMethod>& integrationMethods,
-                         const std::vector<WSTRING>& enabledIntegrationNames,
-                         const std::vector<WSTRING>& disabledIntegrationNames);
+                         const std::vector<WSTRING>& enabledTraceIntegrationNames,
+                         const std::vector<WSTRING>& disabledTraceIntegrationNames,
+                         const std::vector<WSTRING>& enabledLogIntegrationNames,
+                         const std::vector<WSTRING>& disabledLogIntegrationNames);
```
Comment (pellared) [inline src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.h:52]: This is getting very nasty. I think it is time to refactor and introduce a `Integrations` class so that we can change the signatures to something like:

```
Integrations LoadIntegrationsFrom*(inputs...)
```

or even instead of functions create static methods in `Integrations` class so that the are named e.g. `FromJson` or `LoadJson`


We can do it in a separate PR to make it easier for reviewing.

File: docs/config.md [hunk 3]
```diff
@@ -25,10 +25,12 @@ for more details.
...
+| `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS`  | Comma-separated list of metrics source instrumentations you want to enable. Set to `none` to disable all metric instrumentations.                                                                                                                               | all available instrumentations |
 | `OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS` | Comma-separated list of metrics source instrumentations you want to disable.                                                                                                                                                                                    |                                |
+| `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS`     | Comma-separated list of logs source instrumentations you want to enable. Set to `none` to disable all metric instrumentations.                                                                                                                                  | all available instrumentations |
```
Comment (pellared) [inline docs/config.md:32]: Can you please add a test in `SmokeTests` similar to `SubmitLogs` that would set `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS=none` and assert that none logs are send using `collector.AssertEmpty(5.Seconds());`? 

It would test the `OTEL_DOTNET_AUTO_LOGS_ENABLED_INSTRUMENTATIONS` env var. 

Similar tests should be added for traces and metrics (as a separate PR).

I want to have this smoke test added as part of this PR to have any integration tests verifying the new functionality.

File: docs/config.md [hunk 4]
```diff
@@ -25,10 +25,12 @@ for more details.
...
+| `OTEL_DOTNET_AUTO_LOGS_DISABLED_INSTRUMENTATIONS`    | Comma-separated list of logs source instrumentations you want to disable.                                                                                                                                                                                       |                                |
```
[the remark this answers is not in the document]
Comment (pellared) [reply docs/config.md:33]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1480

File: docs/config.md (hunk 3 above)
Comment (pellared) [reply docs/config.md:32]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1481

File: src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.h (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.h:52]: I agree that it could be done better, tracked under #1482

File: src/OpenTelemetry.AutoInstrumentation/Instrumentations/InstrumentationType.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations;
+
+internal enum InstrumentationType
+{
+    Trace,
+    // Metric,
```
[the remark this answers is not in the document]
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentations/InstrumentationType.cs:22]: It should be covered by #696

**Review by dszmigielski — APPROVED** [review]

File: docs/config.md (hunk 3 above)
Comment (Kielek) [reply docs/config.md:32]: a10cd338bf12a1d8b69cf7bae47729711e1c6b9b

**Review by pellared — APPROVED** [review]

**Review by RassK — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]
